frontend-react/src/pages/admin/AdminRevHistory.tsx

```diff
@@ -0,0 +1,273 @@
+import { Helmet } from "react-helmet";
+import React, { Suspense, useCallback, useState } from "react";
+import { Grid, GridContainer, Accordion } from "@trussworks/react-uswds";
+import { AccordionItemProps } from "@trussworks/react-uswds/lib/components/Accordion/Accordion";
+import { useParams } from "react-router-dom";
+
+import HipaaNotice from "../../components/HipaaNotice";
+import {
+    SettingRevision,
+    SettingRevisionParams,
+    useSettingRevisionEndpointsQuery,
+} from "../../network/api/Organizations/SettingRevisions";
+import Spinner from "../../components/Spinner";
+import { AuthElement } from "../../components/AuthElement";
+import { MemberType } from "../../hooks/UseOktaMemberships";
+import { jsonSortReplacer } from "../../utils/JsonSortReplacer";
+import { formatDate } from "../../utils/misc";
+import { StaticCompare } from "../../components/StaticCompare";
+
+/** Accordion components need data in specific format **/
+const dataToAccordionItems = (props: {
+    key: string; // used for React key and passed back to the onClickHandler
+    selectedKey: string;
+    onClickHandler: (
+        key: string,
+        itemClickedKey: string,
+        data: SettingRevision
+    ) => void;
+    data: SettingRevision[];
+}): AccordionItemProps[] => {
+    const results: AccordionItemProps[] = [];
+    if (props.data.length === 0) {
+        return [];
+    }
+
+    // should come back sorted by name and version from server. Sort by name, then by version
+    props.data.sort((a, b) =>
+        a.name === b.name ? a.version - b.version : a.name.localeCompare(b.name)
+    );
+
+    // group the data to make generating the prop content logic easier to follow
+    const grouped: { [key: string]: SettingRevision[] } = {};
+    props.data.forEach((item) => {
```

Comment:
would prefer a reducer implementation

Reply:
Fair. I'll rewrite.

Reply:
Wait `array.reduce` or `useReducer`? There's non-constant logic above it so I assume you mean some array level operation?

Reply:
yeah, I'm just talking about Array.reduce to avoid an unnecessary closure. feel free to tell me to bug off

Reply:
OK. This is a _common_ problem. I found a stock reducer that can be used to d a groupBy that was in typescript and added it to misc (with tests).